client.go

```diff
@@ -0,0 +1,293 @@
+	}
+
+	if strings.HasSuffix(uc.options.url, deprecatedSuffix) {
+		uc.warn(fmt.Errorf("Unleash server URL %s should no longer link directly to /features", uc.options.url))
+		uc.options.url = strings.TrimSuffix(uc.options.url, deprecatedSuffix)
+	}
+
+	if !strings.HasSuffix(uc.options.url, "/") {
+		uc.options.url += "/"
+	}
+
+	parsedUrl, err := url.Parse(uc.options.url)
+	if err != nil {
+		return nil, err
+	}
+
+	if uc.options.appName == "" {
+		return nil, fmt.Errorf("Unleash client appName missing")
+	}
+
+	if uc.options.instanceId == "" {
+		uc.options.instanceId = generateInstanceId()
+	}
+
+	uc.repository = newRepository(
+		repositoryOptions{
+			backupPath:      uc.options.backupPath,
+			url:             *parsedUrl,
+			appName:         uc.options.appName,
+			instanceId:      uc.options.instanceId,
+			refreshInterval: uc.options.refreshInterval,
+			storage:         uc.options.storage,
+			httpClient:      uc.options.httpClient,
+		},
+		repositoryChannels{
+			errorChannels: errChannels,
+			ready:         uc.ready,
+		},
+	)
+
+	uc.strategies = append(defaultStrategies, uc.options.strategies...)
+
+	strategyNames := make([]string, len(uc.strategies))
+	for i, strategy := range uc.strategies {
+		strategyNames[i] = strategy.Name()
+	}
+
+	uc.metrics = newMetrics(
+		metricsOptions{
+			appName:         uc.options.appName,
+			instanceID:      uc.options.instanceId,
+			strategies:      strategyNames,
+			metricsInterval: uc.options.metricsInterval,
+			url:             *parsedUrl,
+			httpClient:      uc.options.httpClient,
+		},
+		metricsChannels{
+			errorChannels: errChannels,
+			count:         uc.count,
+			sent:          uc.sent,
+			registered:    uc.registered,
+		},
+	)
+
+	return uc, nil
+}
+
+func (uc *Client) sync() {
+	for {
+		select {
+		case e := <-uc.errors:
+			if uc.errorListener != nil {
+				uc.errorListener.OnError(e)
+			}
+		case w := <-uc.warnings:
+			if uc.errorListener != nil {
+				uc.errorListener.OnWarning(w)
+			}
+		case <-uc.ready:
+			if uc.repositoryListener != nil {
+				uc.repositoryListener.OnReady()
+			}
+		case m := <-uc.count:
+			if uc.metricsListener != nil {
+				uc.metricsListener.OnCount(m.Name, m.Enabled)
+			}
+		case md := <-uc.sent:
+			if uc.metricsListener != nil {
+				uc.metricsListener.OnSent(md)
+			}
+		case cd := <-uc.registered:
+			if uc.metricsListener != nil {
+				uc.metricsListener.OnRegistered(cd)
+			}
+		case <-uc.closed:
+			return
+		}
+	}
+}
+
+// IsEnabled queries whether or not the specified feature is enabled or not.
+func (uc Client) IsEnabled(feature string, options ...FeatureOption) (enabled bool) {
+	defer func() {
+		uc.metrics.count(feature, enabled)
+	}()
+
+	f := uc.repository.getToggle(feature)
+
+	var opts featureOption
+	for _, o := range options {
+		o(&opts)
+	}
+
+	if f == nil {
+		if opts.fallback != nil {
+			return *opts.fallback
+		}
+		return false
+	}
+
+	if !f.Enabled {
+		return false
+	}
+
+	for _, s := range f.Strategies {
```

Comment:
A feature toggle can have multiple strategies and one of them could be enabled, but not necessary the first one. Thus you have to continue if the first found strategy returns false until there is no more strategies.